src/ui_components/settings.js

```diff
@@ -166,6 +166,16 @@ class Settings extends React.Component {
                                 onChange={this.handleChangeZoomMap} />
                         </div>
                     </Ons.ListItem>
+                    <Ons.ListItem key='locationPublic'>
+                        <div className='left'>
+                            <p>Share location</p>
+                        </div>
+                        <div className='right'>
+                            <Ons.Switch
+                                checked={this.props.locationPublic}
+                                onChange={this.handleLocationPublicChange} />
```

Comment:
Oh, just saw your comment about the log, I think it's because this method isn't defined here in settings.js. Don't you need to define it here, something like this?
https://github.com/lbraun/geofreebie/blob/master/src/ui_components/settings.js#L99-L103

And then pass the onLocationPublicChange method into the settings component, like we do here?
https://github.com/lbraun/geofreebie/blob/master/src/ui_components/app.js#L216